uamqp/management_link.py

```diff
@@ -164,18 +179,36 @@ def execute_operation(
             type=None,
             locales=None
     ):
+        message.application_properties["operation"] = operation
+        message.application_properties["type"] = type
+        message.application_properties["locales"] = locales
+        try:
+            # TODO: namedtuple is immutable, which may push us to re-think about the namedtuple approach for Message
+            new_properties = message.properties._replace(message_id=self.next_message_id)
+        except AttributeError:
+            new_properties = Properties(message_id=self.next_message_id)
+        message = message._replace(properties=new_properties)
 
         self._request_link.send_transfer(
             message,
             on_send_complete=self._on_send_complete,
             timeout=timeout
         )
-        self._pending_operations.append(PendingMgmtOperation(message, on_execute_operation_complete))
+        self.next_message_id += 1
+        self._pending_operations.append(PendingManagementOperation(message, on_execute_operation_complete))
```

Comment:
Where's is this PendingManagementOperation defined?

Reply:
it's a namedtuple defined at the top of this file
https://github.com/Azure/azure-uamqp-python/blob/pyproto/uamqp/management_link.py#L24